atom/browser/atom_download_manager_delegate.cc

```diff
@@ -139,7 +139,12 @@ void AtomDownloadManagerDelegate::OnDownloadPathGenerated(
   if (path.empty()) {
     std::vector<base::FilePath::StringType> extensions;
     base::FilePath::StringType extension;
-    if (GetItemExtension(item, &extension)) {
+    if (!GetItemExtension(item, &extension)) {
+      extension = target_path.Extension();
```

Comment:
wouldn't it make more sense to put this code in `GetItemExtension`?

Reply:
yeah, good call. will do a follow up